tempesta_fw/cache.c

```diff
@@ -190,6 +191,73 @@ tfw_cache_msg_cacheable(TfwHttpReq *req)
 	return __cache_method_test(req->method);
 }
 
+/*
+ * Find caching policy in specific vhost and location.
+ */
+static int
+tfw_cache_policy(TfwVhost *vhost, TfwLocation *loc, TfwStr *arg)
+{
+	TfwCaPolicy *capo;
+
+	/* Search locations in current vhost. */
+	if (loc && loc->capo_sz) {
+		if ((capo = tfw_capolicy_match(loc, arg)))
+			return capo->cmd;
+	}
+
+	/*
+	 * Search default policies in current vhost.
+	 * If there's none, then search global default policies.
+	 */
+	loc = vhost->loc_dflt;
+	if (loc && loc->capo_sz) {
+		if ((capo = tfw_capolicy_match(loc, arg)))
+			return capo->cmd;
+	} else {
+		TfwVhost *vhost_dflt = tfw_vhost_get_default();
+		if (vhost == vhost_dflt)
+			return TFW_D_CACHE_BYPASS;
+
```

Comment:
Please comment here or at `tfw_vhost_dflt` initialization that `loc_dflt` is always global default caching policy. This place made me think about couple of minutes what's going on. This is tricky place which we have discussed a lot.